### PR TITLE
Add webhook metadata api integration tests

### PR DESCRIPTION
**Pull request by Sachin-Mamoru**
This pull request introduces a comprehensive suite of integration tests for the Webhook Metadata REST API, covering both successful and failure scenarios. The changes include new test classes for validating the API's happy paths and negative paths, a shared test base for common setup and utilities, and updates to the test configuration to ensure these tests are executed.

**Webhook Metadata REST API Integration Tests**

*New test classes:*
- Added `WebhookMetadataSuccessTest` to verify successful responses and data structure for the Webhook Metadata API, including checks for profiles, adapter info, and specific event channels.
- Added `WebhookMetadataFailureTest` to validate error handling for invalid profile requests, ensuring correct error codes and messages are returned.

*Test infrastructure:*
- Introduced `WebhookMetadataTestBase` to provide shared constants, setup logic, and utility methods for the new test classes, improving code reuse and maintainability.

*Test configuration:*
- Updated `testng.xml` to include the new Webhook Metadata test classes in the integration test suite, ensuring they run as part of the automated test pipeline.

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive tests for webhook metadata REST APIs: successful retrieval of webhook metadata and event profile details.
  * Added negative-path tests validating proper 404 error responses for invalid profile names.
  * Introduced shared test setup/utilities and lifecycle hooks to standardize webhook metadata testing.
  * Expanded the REST API test suite configuration to include the new webhook metadata tests.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/webhook/metadata/v1/WebhookMetadataSuccessTest.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.notNullValue;
+
+/**
+ * Tests for happy paths of the Webhook Management REST API.
```
[the remark this answers is not in the document]
Comment (Sachin-Mamoru) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/webhook/metadata/v1/WebhookMetadataSuccessTest.java]: fixed with https://github.com/wso2/product-is/pull/26373/commits/ecd8faf1f896d3c312beccb19e0e448adc08ed90

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/webhook/metadata/v1/WebhookMetadataFailureTest.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.Matchers.notNullValue;
+
+/**
+ * Negative-path tests for the Webhook Management REST API.
```
[the remark this answers is not in the document]
Comment (Sachin-Mamoru) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/webhook/metadata/v1/WebhookMetadataFailureTest.java]: fixed with https://github.com/wso2/product-is/pull/26373/commits/ecd8faf1f896d3c312beccb19e0e448adc08ed90

**Review by ashanthamara — APPROVED** [review]
